### PR TITLE
Integrate QBFT into Besu

### PR DESCRIPTION
**Pull request by jframe**
Signed-off-by: Jason Frame <jasonwframe@gmail.com>

## PR description
Allow Besu to use QBFT consensus mechanism. The configuration uses the same values as IBFT2 but under the qbft namespace instead ibft. Also adds a basic mining acceptance test for QBFT.

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: acceptance-tests/tests/src/test/resources/log4j2.xml [hunk 1]
```diff
@@ -77,7 +77,7 @@
 
   </Appenders>
   <Loggers>
-    <Logger name="org.hyperledger.besu.SubProcessLog" level="INFO" additivity="false">
+    <Logger name="org.hyperledger.besu.SubProcessLog" level="TRACE" additivity="false">
```
[the remark this answers is not in the document]
Comment (jframe) [reply acceptance-tests/tests/src/test/resources/log4j2.xml]: no. that change should be gone now

File: acceptance-tests/tests/src/test/resources/qbft/qbft.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
```
[the remark this answers is not in the document]
Comment (jframe) [reply acceptance-tests/tests/src/test/resources/qbft/qbft.json:1]: prefer that there was a seperate qbft config file to keep this simple. if we want to go with a common bft format that could be a future PR

[the remark this answers is not in the document]
Comment (rain-on) [reply acceptance-tests/tests/src/test/resources/qbft/qbft.json:1]: 👍 sounds good

File: besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        new BftBlockCreatorFactory(
+            gasLimitCalculator,
+            transactionPool.getPendingTransactions(),
+            protocolContext,
+            protocolSchedule,
+            miningParameters,
+            localAddress,
+            bftConfig.getMiningBeneficiary().map(Address::fromHexString).orElse(localAddress));
+
+    // NOTE: peers should not be used for accessing the network as it does not enforce the
```
Comment (rain-on) [inline besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java:131]: can this comment be removed? I mean, its sensible, but its not useful right here.

Comment (rain-on) [reply besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java:131]: maybe more useful on line 139/140?

**Review by rain-on — APPROVED** [review]
LGTM - maybe 1 bad comment to be moved/removed.

File: besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java (hunk 3 above)
Comment (jframe) [reply besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java]: done

**Pull request merged**